Apromore-Core-Components/Apromore-Manager/src/test/java/org/apromore/service/impl/UserMetadataBuilder.java

```diff
@@ -0,0 +1,34 @@
+package org.apromore.service.impl;
+
+import org.apromore.dao.model.*;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class UserMetadataBuilder {
+
+    Group group;
+    UsermetadataType usermetadataType;
+    Usermetadata usermetadata;
+
+    Set<GroupUsermetadata> groupUsermetadataSet = new HashSet<>();
+    Set<UsermetadataLog> usermetadataLogSet = new HashSet<>();
+    Set<UsermetadataProcess> usermetadataProcessSet = new HashSet<>();
+
+    public UserMetadataBuilder() {
+
+    }
+
+    public UserMetadataBuilder withUsermetadataType(Integer id, String type){
+        usermetadataType = new UsermetadataType();
+        usermetadataType.setIsValid(true);
+        usermetadataType.setType(type);
+        usermetadataType.setVersion(1);
+        usermetadataType.setId(id);
+        return this;
+    }
+
+    public UsermetadataType buildUsermetadataType() {
+        return usermetadataType;
+    }
+}
```

Comment:
Don't get these two methods. If you need to create a UsermetadataType from an "id" and "type", it can be done with buildUsermetadataType(Integer id, String type), why you need two calls: withUsermetadataType(Integer id, String type) and then buildUsermetadataType() with no parameters? In addition, buildUsermetadataType can be a static method if this class is only to create a UsermetadataType from an id and type param and so the caller won't need to instantiate any UserMetadataBuilder object, just call UserMetadataBuilder.buildUsermetadataType.